lib/sdn_epc/supervisor.ex

```diff
@@ -0,0 +1,28 @@
+defmodule SdnEpc.Supervisor do
+  use Supervisor
+
+  def start_link do
+    res = Supervisor.start_link __MODULE__, [], name: __MODULE__
+    start_ofp_channel_sup()
+    start_worker()
+    res
+  end
+
+  def start_ofp_channel_sup do
+    ofpc_sup_spec = supervisor(:ofp_channel_sup, [1])
+    {:ok, ofpc_sup } = Supervisor.start_child __MODULE__, ofpc_sup_spec
+    Process.register ofpc_sup, :ofp_channel_sup
+  end
+
+  def start_worker do
+    Supervisor.start_child __MODULE__, worker(SdnEpc.Forwarder, []) 
```

Comment:
This is not correct. The worker should be started by the supervisor itself and the specification should be provided in the `init/1`.

Actually, there's should be separate `simple_one_for_one` supervisor only for channels and the top level sup of your app would supervise the Forwarder and that supervisor.

Then both of these processes would be started through specification and the the `channel_sups` would be started dynamically on an event of a switch connecting, probably somehow indirectly from `ofsh_calls`.